lib/ansible/modules/extended_mount_facts.py

```diff
@@ -0,0 +1,300 @@
+    description:
+      - This is the maximum number of seconds to query for each mount. This can also be set to V(null) to wait indefinitely.
+      - This timeout also applies to the C(mount) command, which is executed if mounts cannot be listed using a file.
+    required: true
+    type: raw
+  on_timeout:
+    description:
+      - The action to take when a timeout occurs.
+      - When this is set to V(warn) or V(ignore), mounts that time out will be excluded from the results.
+    choices:
+      - error
+      - warn
+      - ignore
+    default: error
+    type: str
+extends_documentation_fragment:
+  - action_common_attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+  platform:
+    platforms: posix
+author:
+  - Ansible Core Team
+"""
+
+EXAMPLES = """
+- name: Get NFS and FUSE subtype mounts
+  extended_mount_facts:
+    devices:
+      - "[!/]*"
+    fstypes:
+      - "nfs"
+      - "nfs4"
+      - "fuse.*"
+"""
+
+RETURN = """
+ansible_facts:
+    description: An ansible_facts dictionary containing C(extended_mounts).
+    returned: always
+    type: dict
+    sample:
+      extended_mounts:
+        /mnt/mount:
+          info:
+            block_available: 3242510
+            block_size: 4096
+            block_total: 3789825
+            block_used: 547315
+            device: hostname:/srv/sshfs
+            fstype: fuse.sshfs
+            inode_available: 1875503
+            inode_total: 1966080
+            mount: /mnt/mount
+            options: "rw,nosuid,nodev,relatime,user_id=0,group_id=0"
+            size_available: 13281320960
+            size_total: 15523123200
+          uuid: N/A
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.common.validation import check_type_int
+from ansible.module_utils.facts import timeout
+from ansible.module_utils.facts.hardware.linux import LinuxHardware
+from ansible.module_utils.facts.utils import get_mount_size, get_file_content
+
+from fnmatch import fnmatch
+from typing import Callable, Optional
+from sys import platform
+
+import re
+import subprocess
+
+# Regular expressions to match individual lines from mount
+
+# FreeBSD: hostname:/srv/sshfs on /mnt/sshfs (fusefs.sshfs)
+# MacOSX: //user@hostname/public on /mnt/cifs (smbfs, nodev, nosuid, mounted by user)
+BSD_MOUNT_RE = re.compile(r"^(?P<device>\S+) on (?P<mount>\S+) \((?P<fstype>.+)\)$")
+
+# AIX: hostname /mount /mounted_over nfs Dec 17 08:06 ro, log  =/dev/hd8
+AIX_MOUNT_RE = re.compile(r"^(?P<node>\S+)\s+(?P<mount>\S+)\s+(?P<mounted_over>\S+)\s+(?P<fstype>\S+)\s+(?P<time>\S+\s+\d+\s+\d+:\d+)\s+(?P<options>.*)$")
+
+# TODO...?
+
+# No mount parsing preferably
+# Fedora, Debian, Ubuntu, Arch use /etc/mtab or /proc/mounts
+# Solaris uses /etc/mnttab
+
+
+def get_mount_info(
+    module: AnsibleModule,
+    mount: str,
+    device: str,
+    uuids: dict[str, str],
+    udevadm_uuid: Callable[[str], str]
+) -> tuple[Optional[dict], Optional[str]]:
```

Comment:
Nit: `Optional` is considered confusing, so many linters recommend being explicit about its actual semantics:
```suggestion
) -> tuple[dict | None, str | None]:
```

Reply:
Thanks, I forgot about that.

Reply:
Fixed